File: meldingsvarsler/src/main/java/no/sonat/meldingsvarsler/SMSVarsler.java
package no.sonat.meldingsvarsler;

import java.io.*;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class SMSVarsler {
    private List<String> abonnenter;
    private String melding;
    public void leggTilbonnenter(List<String> abonnenter){
        this.abonnenter = abonnenter;
    }

    public void leggTilAbonnent(String abonnent){
        this.abonnenter.add(abonnent);
    }

    public void sendMelding(final String melding) {
        abonnenter.forEach(abonnent -> sendSms(abonnent, melding));
    }

    private void sendSms(String abonnent, String melding) {
        System.out.println("TLF: " + abonnent + " " + melding);
    }

    public void hentAbonnenter() {
        File file = new File("abonnenter.csv");
        try (
            BufferedReader reader = new BufferedReader(new FileReader(file))) {
            String line;
            StringBuffer buffer = new StringBuffer();
            while((line = reader.readLine()) != null) {
                buffer.append(line);
            }
            this.abonnenter = Arrays.asList(buffer.toString().split(","));
        } catch (IOException e) {
        }
    }

    public void lagreAbonnenter() throws IOException {
        File file = new File("abonnenter.csv");
        FileWriter fileWriter = null;
        try {
            fileWriter = new FileWriter(file);
            fileWriter.write(abonnenter.stream().collect(Collectors.joining(",")));
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            try {
                fileWriter.close();
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }
}

File: meldingsvarsler/App/SMSVarsler.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace App {
    public class SmsVarsler 
    {
        List<string> abonnenter;
        string melding;

        public void LeggTilAbonnenter(List<string> abonnenter){
            this.abonnenter = abonnenter;
        }

        public void LeggTilAbonnent(string abonnent){
            abonnenter.Add(abonnent);
        }

        public void SendMelding(string melding) {
            abonnenter.ForEach(abonnent => SendSms(abonnent, melding));
        }

        void SendSms(string abonnent, string melding) {
            Console.Out.WriteLine("TLF: " + abonnent + " " + melding);
        }

        public void HentAbonnenter() 
        {
            var fileStream = File.OpenRead("abonnenter.csv");
            var reader = new StreamReader(fileStream);
            string line;
            var buffer = new StringBuilder();
            while((line = reader.ReadLine()) != null)
            {
                if (buffer.Length > 0) buffer.Append(",");
                buffer.Append(line);
            }

            abonnenter = buffer.ToString().Split(',').ToList();
        }

        public void LagreAbonnenter()
        {
            var fileStream = File.OpenWrite("abonnenter.csv");
            var fileWriter = new StreamWriter(fileStream);
            fileWriter.Write(string.Join(",", abonnenter));
            fileWriter.Close();
        }
    }
}
